resim/msg/byte_swap_helpers.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 ReSim, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#include "resim/msg/byte_swap_helpers.hh"
+
+#include <cstring>
+#include <utility>
+
+namespace resim::msg {
+
+template <std::endian target_order>
+void set_data(int16_t data, InOut<Int16> msg) {
+  constexpr size_t NUM_BYTES = 2;
+  std::string serialized{NUM_BYTES, '\0'};
+  auto data_bits = std::bit_cast<std::array<std::byte, NUM_BYTES>>(data);
+
+  static_assert(
+      std::endian::native == std::endian::little or
+      std::endian::native == std::endian::big);
+  if constexpr (std::endian::native != target_order) {
+    std::swap(data_bits.at(0), data_bits.at(1));
+  }
+
+  std::memcpy(serialized.data(), &data_bits, NUM_BYTES);
+  msg->set_data(serialized);
```

Comment:
I enjoy that none of this code requires me as a reviewer to remember which is big endian and which is little endian.

Reply:
This also helps make sure we cover this branch while only running on a little endian system. 😅